### PR TITLE
Fix exception traceback to omit superfluous 'Backtrace' information

### LINKED ISSUE #9977: Selenium 4 exceptions produce a Traceback that contains this superfluous 'Backtrace' info.
### What happened?

Selenium 4 exceptions produce a Traceback that contains superfluous 'Backtrace' information that is not useful for ordinary debugging as the Traceback already contains the traditional call stack information.

### How can we reproduce the issue?

```shell
Create an exception that produces a ElementClickInterceptedException

   def webwait_rl(self, locator_type, locator_value, condition=EC.element_to_be_clickable, wait=wait_default):
        try:
            result = WebDriverWait(self.browser, wait).until(condition((self.locator_map[locator_type], locator_value)))
            return result
        except Exception as e:
            raise AssertionError("\n\n%s\nwebwait_rl exception %s\n%r\nlocator: %s=%s   wait: %d   condition: %s\n%s"
                                 % (self.output_stack(ss), e, self.e_type(e), locator_type, locator_value, wait,
                                    condition, msg))
```

### Relevant log output

```shell
Selenium 3.141.0
FAILED (failures=1)
FailureTraceback (most recent call last):
  File "C:\Users\pflaherty\pycharmprojects\automation-master\Live\RCV_Guardian\RcvGuardian_Tip.py", line 247, in test_001_create_guardian_eyewitness_tip
    rcv_main.close_module_by_name("SAFETY_PROFILE")
  File "C:\Users\pflaherty\pycharmprojects\automation-master\RCV_Guardian_Pages\RcvGuardianMainPage.py", line 368, in close_module_by_name
    self.click_(x_query, retry=3, msg=f"Unable to click button to close the '{module_name}' module.")
  File "C:\Users\pflaherty\pycharmprojects\automation-master\PageObject\AWDTools.py", line 293, in click_
    (msg, locator, i, self.output_stack(), ex))
AssertionError: Unable to click button to close the 'SAFETY_PROFILE' module.
click_ failed using locator xpath=//*[@smartlettype='SAFETY_PROFILE']//div[@class='action smartletClose']/a after 3 retries
test_001_create_guardian_eyewitness_tip--close_module_by_name--click_--491-3783
10-22-21  10:43:03AMElementClickInterceptedException('element click intercepted: Element <a>...</a> is not clickable at point (1321, 321). Other element would receive the click: <div id="chatConversation" class="textPosterTarget eyewitness">...</div>\n  (Session info: chrome=94.0.4606.81)', None, None)

Selenium 4.0
FAILED (failures=1)
FailureTraceback (most recent call last):
  File "C:\Users\pflaherty\pycharmprojects\automation-master\Live\RCV_Guardian\RcvGuardian_Tip.py", line 247, in test_001_create_guardian_eyewitness_tip
    rcv_main.close_module_by_name("SAFETY_PROFILE")
  File "C:\Users\pflaherty\pycharmprojects\automation-master\RCV_Guardian_Pages\RcvGuardianMainPage.py", line 368, in close_module_by_name
    self.click_(x_query, retry=3, msg=f"Unable to click button to close the '{module_name}' module.")
  File "C:\Users\pflaherty\pycharmprojects\automation-master\PageObject\AWDTools.py", line 293, in click_
    (msg, locator, i, self.output_stack(), ex))
AssertionError: Unable to click button to close the 'SAFETY_PROFILE' module.
click_ failed using locator xpath=//*[@smartlettype='SAFETY_PROFILE']//div[@class='action smartletClose']/a after 3 retries
test_001_create_guardian_eyewitness_tip--close_module_by_name--click_--491-3161
10-22-21  10:32:41AMElementClickInterceptedException('element click intercepted: Element <a>...</a> is not clickable at point (1321, 321). Other element would receive the click: <div id="chatConversation" class="textPosterTarget eyewitness">...</div>\n  (Session info: chrome=94.0.4606.81)', None, ['Backtrace:', '\tOrdinal0 [0x0080BDE3+2473443]', '\tOrdinal0 [0x007A6661+2057825]', '\tOrdinal0 [0x006B2438+1057848]', '\tOrdinal0 [0x006E27B9+1255353]', '\tOrdinal0 [0x006E0AE0+1247968]', '\tOrdinal0 [0x006DE8AD+1239213]', '\tOrdinal0 [0x006DD701+1234689]', '\tOrdinal0 [0x006D3567+1193319]', '\tOrdinal0 [0x006F56E3+1332963]', '\tOrdinal0 [0x006D3386+1192838]', '\tOrdinal0 [0x006F57CA+1333194]', '\tOrdinal0 [0x00704ABF+1395391]', '\tOrdinal0 [0x006F55AB+1332651]', '\tOrdinal0 [0x006D2104+1188100]', '\tOrdinal0 [0x006D2F59+1191769]', '\tGetHandleVerifier [0x00992266+1549718]', '\tGetHandleVerifier [0x00A3D4A7+2250711]', '\tGetHandleVerifier [0x0089718B+521403]', '\tGetHandleVerifier [0x00896229+517465]', '\tOrdinal0 [0x007AB79D+2078621]', '\tOrdinal0 [0x007AFB58+2095960]', '\tOrdinal0 [0x007AFC92+2096274]', '\tOrdinal0 [0x007B9541+2135361]', '\tBaseThreadInitThunk [0x7725FA29+25]', '\tRtlGetAppContainerNamedObjectPath [0x77A47A9E+286]', '\tRtlGetAppContainerNamedObjectPath [0x77A47A6E+238]', ''])
```

### Operating System

Windows 10

### Selenium version

4.0.0

### What are the browser(s) and version(s) where you see this issue?

94.0.4606.81 (Official Build) (64-bit) (cohort: Stable)

### What are the browser driver(s) and version(s) where you see this issue?

ChromeDriver 94.0.4606.61

### Are you using Selenium Grid?

No

### PR DESCRIPTION
**Pull request by SmartManoj**
### **User description**
Fixes #9977

Filter out superfluous 'StackTrace' information from Selenium 4 exception tracebacks.

* Add a check for an environment variable to conditionally include the stacktrace.

---

For more details, open the [Copilot Workspace session](https://copilot-workspace.githubnext.com/SeleniumHQ/selenium/pull/14975?shareId=eb869ea3-8228-4a43-9e2b-2df9e015161c).

___

### **PR Type**
Bug fix

___

### **Description**
- Addresses issue #9977 where Selenium 4 exceptions contain superfluous 'Backtrace' information
- Added environment variable `INCLUDE_STACKTRACE` to control stacktrace visibility in exceptions
- Implemented stacktrace filtering to remove unnecessary 'Backtrace' information
- Modified exception handling to make stack traces more concise and readable
- Changes maintain backward compatibility through environment variable toggle

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>exceptions.py</strong><dd><code>Add environment variable control for stacktrace display</code>&nbsp; &nbsp; </dd></summary>
<hr>

py/selenium/common/exceptions.py

<li>Added import of <code>os</code> module<br> <li> Modified <code>__str__</code> method to only include stacktrace when <br><code>INCLUDE_STACKTRACE</code> environment variable is set

</details>

  </td>
  <td><a href="https://github.com/SeleniumHQ/selenium/pull/14975/files#diff-1b2829f22ba2d98a53d9f003d5f04dbfe70ff4447409c32dcc475fca0cdec754">+2/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></td></tr><tr><td><strong>Bug fix</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>errorhandler.py</strong><dd><code>Implement stacktrace filtering functionality</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

py/selenium/webdriver/remote/errorhandler.py

<li>Added import of <code>os</code> module<br> <li> Added new <code>filter_stacktrace</code> method to remove 'Backtrace' information<br> <li> Modified <code>check_response</code> to filter stacktrace when <code>INCLUDE_STACKTRACE</code> <br>is not set

</details>

  </td>
  <td><a href="https://github.com/SeleniumHQ/selenium/pull/14975/files#diff-3a2cce6f647dc882c5111a63c86a273646543e70b1ba3538181aee04a9385d78">+9/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**: Comment `/help "your question"` on any pull request to receive relevant information

### CODE REVIEW

**Comment by diemol** [comment]
Thanks for rasing the PR, however we commented the [code was going to stay as it was,](https://github.com/SeleniumHQ/selenium/issues/9977#issuecomment-1001114642) is there any other motivation for this?

**Comment by SmartManoj** [comment]
Those stack traces are not useful for SLMs.

File: py/selenium/common/exceptions.py [hunk 1]
```diff
@@ -38,7 +39,7 @@ def __str__(self) -> str:
         exception_msg = f"Message: {self.msg}\n"
         if self.screen:
             exception_msg += "Screenshot: available via screen\n"
-        if self.stacktrace:
+        if self.stacktrace and os.getenv("INCLUDE_STACKTRACE"):
```
Comment (SmartManoj) [inline py/selenium/common/exceptions.py:42]:
```suggestion
        if self.stacktrace and not os.getenv("EXCLUDE_STACKTRACE"):
```
Will this change be accepted?

**Comment by diemol** [comment]
You should be able to exclude those or train the SLM to ignore them. We don't want to change the code as we pass along the information the browser sends back. You should engage more with the Chrome folks in the open issue in their bug tracker.

**Comment by SmartManoj** [comment]
When will these stack traces be useful?
Until the Chrome folks fix that, could you add this as a temporary fix?

**Pull request closed without merging**